### DISCUSSION TITLE
BLDC motor control

### DISCUSSION THREAD

**Discussion by samuk** [Roger-random/Sawppy_Rover#30] [General]
I didn't notice Discussions were open, so opened an issue for it

This might be a better place to discuss https://github.com/Roger-random/Sawppy_Rover/issues/27

**Comment by Roger-random** [comment]
Recap from #27: @samuk wants to investigate adopting brushless DC motor control as an option for Sawppy. A few other Sawppy rover builders are interested in the idea.

I see no fundamental problems with the general idea and have no objections to an investigation. However, it will not be my primary focus in the immediate future due to availability of both time and money.

Let's continue discussion and post investigation updates here. Development work will occur in @samuk's fork of this repository. (https://github.com/samuk/Sawppy_Rover) Once a BLDC Sawppy is up and running we can discuss how to incorporate the fork into main.

**Reply by samuk** [reply]
Thanks Roger. 

My personal interest is to explore this using open hardware driver boards. Which AFAIK means either [NearZero2 ](https://skysedge.com/robotics/nz2/index.html) or one of the many [SimpleFOC](https://community.simplefoc.com/search?expanded=true&q=new%20board) designs. It may be possible to develop something custom for Sawppy, although components are a nightmare at the moment.

Dengs FOC is one [open hardware board](https://github.com/ToanTech/Deng-s-foc-controller) currently [available on Aliexpress](https://www.aliexpress.com/item/1005003311241556.html)

As part of another project I'm helping out on a [~5amp SimpleFOC board in M5stack profile](https://github.com/rosmo-robot/Rosmo_ESC), so I'll probably play with that first.

Odrive is certainly possible too, but I'm personally less interested in that since they closed their hardware and it's expensive.

In terms of motors, I haven't really done any investigation at this scale. Personally, I'd want to be somewhere in the middle of the cheap <> good spectrum.

Perhaps something from [Cubemars](https://store.cubemars.com/goods.php?id=953), [Tmotor](http://www.t-motor.com/en/Product/index/classid/11.html), or [DYS](http://www.dys.hk/product/Gimbal%20Brushless%20Motor.html)

**Reply by Roger-random** [reply]
I'm looking forward to your investigation's findings. The more rovers, and more options for them, the merrier!

As for sharing your modified Sawppy, please put your information in the `/modifications` folder. (https://github.com/Roger-random/Sawppy_Rover/tree/main/modifications)

**Reply by samuk** [reply]
It seems Tenacity is running[ GM25-370](https://www.banggood.com/GM25-370-Motor-DC-6_012_0V-Black-Brushed-Motor-4mm-Shaft-Diameter-Motor-p-1104564.html?cur_warehouse=CN) and would like more torque, so they seem to be the benchmark to improve on if possible. They provide 

~$6 [ GM25-370](https://www.banggood.com/GM25-370-Motor-DC-6_012_0V-Black-Brushed-Motor-4mm-Shaft-Diameter-Motor-p-1104564.html?cur_warehouse=CN) 1.5kg.cm (0.15N.M) at 6v. weigh 100g with brushes and gears.

~$14 [Lx16a](https://www.buildlog.net/blog/2017/09/lewansoul-lobot-lx-16a-serial-servo-review/) 17 kg-cm 45.22mmx24.72mmx36.3mm

~$22 [Hiwonder LX-224HV](https://www.hiwonder.hk/products/hiwonder-lx-224-hv-three-connectors-bus-high-voltage-servo) 20 kg-cm

$45[ BGM5208-75 ](http://www.t-motor.com/en/Product/detail/classid/11/id/73.html) 1950g @5V&0.47A weighing 180g brushless, no gears.

$45 [IP65 Kingmax SW22HV 7T](https://www.aliexpress.com/item/1005002297372748.html) 360degree 22kg/cm wheel 20.20x 40.5 x 38. 270mm cable

$28 [**IP65** FEICHAO 40kg ](https://www.aliexpress.com/item/1005001325059005.html) 40kg/cm

$25 [**IP65** epower](https://www.aliexpress.com/item/1005002835590547.html) steer

$33 [IP66 Hiwonder HPS-3527SG](https://www.hiwonder.hk/products/hiwonder-hps-3527sg-35kg-coreless-digital-servo-metal-gear-and-waterpoof-servo-for-robotic-diy-rc-car-control-angle-270?_pos=1&_sid=0e96da2e0&_ss=r) 270 degree 35/Kg-cm steer

[$24 Nema23 brushless](https://s.click.aliexpress.com/e/_A4yNkH) & [$35 90 worm](https://s.click.aliexpress.com/e/_9feXzf)

$75 [**IP65** Stepper with encoder](https://www.omc-stepperonline.com/waterproof-stepper-motor/p-series-ip65-waterproof-nema-23-closed-loop-stepper-motor-1-2nm-169-97oz-in-with-encoder-1000ppr-4000cpr.html) 1.2Nm, 12Kg/cm actuator

 $93 [IP65 Stepper with encoder](https://www.omc-stepperonline.com/waterproof-stepper-motor/p-series-ip65-waterproof-nema-24-closed-loop-stepper-motor-3nm-424-92oz-in-with-encoder-1000ppr-4000cpr.html) 3Nm, 30Kg/cm 

Something from [loopmotion](http://www.loopmotion.com/Brushless-DC-Motor-c90702/)
[$29](http://www.loopmotion.com/nema23-brushless-dc-motor-round-body-with-57mm-square-flange-36vdc-4000rpm-138w-and-184w-p2752831.html) 

[$25](http://www.loopmotion.com/nema23-brushless-dc-motor-60w-125w-or-188w-p2752789.html)

[$19](http://www.loopmotion.com/nema23-brushless-dc-motor-round-body-with-57mm-square-flange-36vdc-4000rpm-46w-and-92w-p2752821.html)

**Reply by Roger-random** [reply]
If you have not done so already, I would like to suggest that you get one or two LewanSoul/HiWonder LX-16A serial bus servos used on Sawppy V1 design. They can serve as a known baseline for comparing these different brushless motors against.
